libs/cache_line/src/lib.rs

```diff
@@ -0,0 +1,151 @@
+#![feature(const_trait_impl)]
+#![no_std]
+
+use core::{mem, ops};
+
+const LINE_SIZE: usize = 64;
+
+/// A cache line optimised array.
+///
+/// It spreads elements across as many cache lines as possible. For example if
+/// only the first four elements of a 256 byte array are being accessed, each
+/// element will have its own cache line.
+///
+/// Indexing is faster if the number of cache lines that the array spreads is a
+/// power of two.
+#[derive(Debug)]
+// Alignment must be the same as `LINE_SIZE`.
+#[repr(align(64))]
+pub struct CacheLineOptimisedArray<T, const N: usize> {
```

Comment:
naming is hard. I feel like "Cache line-optimized array" is a bit vague, specifically "optimized". Can we replace optimized with a different, more specific word that indicates what the array is actually doing to its elements?

possible ideas to get the creative juices flowing: "aligned", "sized", "distributed", "segregated", "placed", "separated". We can bikeshed it a bit if you want